lib/src/data/group_class.dart

```diff
@@ -0,0 +1,47 @@
+import "utils.dart";
+
+enum GroupStatus {
+  openToAll,
+  openByRequest,
+  private,
+}
+
+class Group {
```

Comment:
Let's add back `bool canJoin()`, since it can be useful for the UI people to have. Going through the cases again, 
- open to all --> join
- open by request --> send a request 
- private --> invisible in the UI 
- member limit --> check how many members there are

Needing to send a request doesn't mean this should be `false` though, so this function is really just checking for `private` and the optional member limit. Since private groups shouldn't show up in the UI at all, this function can really just check for `memberLimit == null || members.length < memberLimit`